### DISCUSSION TITLE
Customer Values are replaced by 'XXXXX.' when using /vendor/directFulfillment/orders/v1/purchaseOrders

### DISCUSSION THREAD

**Discussion by ait1ispring** [amzn/selling-partner-api-models#3699] [General]
Problem description:
We were able to use the /vendor/directFulfillment/orders/v1/purchaseOrders to retrieve the information of the Vendor Direct Fulfillment customer, such as name, city, addressLine1, and phone two weeks ago. However, we are now unable to retrieve any customer information; it has been replaced by 'XXXXX.' Anyone has the same situation? We are using the RDT to retrieve the PII.

Error:
There is no error, just want to know if we need to use the RDT to access the information of the customer, since we have already given the RDT access on the SA API.

**Comment by MeenaAmz** [comment]
Have you tried doing the call using RDT as the access token? Customer information is restricted so see if you can retrieve the information using RDT

**Reply by ait1ispring** [reply]
Hi, the response from the Amazon support: "generate a new refresh token for each user and attempt the call again". But I don't understand the "generate a new refresh token for each user". Anyone can provide more details? I did refresh the app token on the Amazon vendor central API console.

**Reply by MeenaAmz** [reply]
As long as you have a access token that is not expired for your app, you should be good. I'm not sure what the guidance from support was related to.
For your issue, couple of questions,
1. Do you have "Direct to Consumer Shipping (Restricted) role" access? 
2. Have you tried calling "/vendor/directFulfillment/orders/v1/purchaseOrders" with restricted data token (RDT) instead of LWA and still not getting the customer information? For more information, please refer to [tutorial](https://developer-docs.amazon.com/sp-api-blog/docs/how-to-test-selling-partner-api-endpoints#step-2-make-a-call-to-the-sp-api-sandbox-endpoint) on how to call SP-API endpoints using RDT. 
I suggest that you address these two to get it resolved.

**Reply by ait1ispring** [reply]
Hi,
1. Yes. I did have the "Direct to Consumer Shipping (Restricted) role" access.
2.  I tried calling "/vendor/directFulfillment/orders/v1/purchaseOrders" with restricted data token (RDT) instead of LWA and still not getting the customer information. Don't know why the response shipping information of the customer is still replaced by the "XXXX"

**Reply by MeenaAmz** [reply]
This seems like an issue specific to your account. Have you tried creating a support case via vendor central?
